lizrd/scripts/grid.py

```diff
@@ -25,92 +24,93 @@
-    TIME = grid_args.get("time", TIME)
-    GRES = grid_args.get("gres", GRES)
-    DRY_RUN = grid_args.get("dry_run", DRY_RUN)
-    SINGULARITY_IMAGE = grid_args.get("singularity_image", SINGULARITY_IMAGE)
-    HF_DATASETS_CACHE = grid_args.get("hf_datasets_cache", HF_DATASETS_CACHE)
-    RUNS_MULTIPLIER = grid_args.get("runs_multiplier", RUNS_MULTIPLIER)
-    INTERACTIVE_DEBUG = grid_args.get("interactive_debug", INTERACTIVE_DEBUG)
-    PUSH_TO_GIT = grid_args.get("push_to_git", PUSH_TO_GIT)
-    NODELIST = grid_args.get("nodelist", NODELIST)
-    N_GPUS = grid_args.get("n_gpus", N_GPUS)
-    CPUS_PER_GPU = grid_args.get("cpus_per_gpu", CPUS_PER_GPU)
-    CUDA_VISIBLE_DEVICES = grid_args.get("cuda_visible", CUDA_VISIBLE_DEVICES)
-
-    if SINGULARITY_IMAGE is None and runner != MachineBackend.LOCAL:
-        raise ValueError("Singularity image is not specified (in JSON or env variable)")
-
-    if NODELIST is not None:
-        NODELIST = "--nodelist=" + NODELIST
-
-    grid = create_grid(PARAMS)
-    grid = multiply_grid(grid, RUNS_MULTIPLIER)
-    no_experiments = len(grid)
-    minutes_per_exp = timestr_to_minutes(TIME)
+
+    for config in configs:
+        config["params"]["path_to_config"] = sys.argv[1]
+
+    grid = []
+    total_no_experiments = 0
+    total_minutes = 0
+
+    for i, grid_args in enumerate(configs):
+        print(f"\nProcessing config {i}...")
+        pprint.pprint(grid_args)
+        RUNNER = grid_args["runner"]
+        PARAMS = grid_args["params"]
+        TIME = grid_args.get("time", "1-00:00:00")
+        GRES = grid_args.get("gres", "gpu:titanv:1")
+        DRY_RUN = grid_args.get("dry_run", False)
+        SINGULARITY_IMAGE = grid_args.get(
+            "singularity_image", get_sparsity_image(runner)
+        )
+        HF_DATASETS_CACHE = grid_args.get("hf_datasets_cache", get_cache_path(runner))
+        RUNS_MULTIPLIER = grid_args.get("runs_multiplier", 1)
+        INTERACTIVE_DEBUG = grid_args.get("interactive_debug", False)
+        NODELIST = grid_args.get("nodelist", None)
+        N_GPUS = grid_args.get("n_gpus", 1)
+        CPUS_PER_GPU = grid_args.get("cpus_per_gpu", 8)
+        CUDA_VISIBLE_DEVICES = grid_args.get("cuda_visible", None)
+
+        if SINGULARITY_IMAGE is None and runner != MachineBackend.LOCAL:
+            raise ValueError(
+                "Singularity image is not specified (in JSON or env variable)"
+            )
+
+        if NODELIST is not None:
+            NODELIST = "--nodelist=" + NODELIST
+
+        PARAMS["temp_args"] = dict()
+        for name, param in zip(
+            [
+                "gres",
+                "time",
+                "n_gpus",
+                "runner",
+                "cpus_per_gpu",
+                "nodelist",
+                "cuda_visible",
+            ],
+            [GRES, TIME, N_GPUS, RUNNER, CPUS_PER_GPU, NODELIST, CUDA_VISIBLE_DEVICES],
+        ):
+            PARAMS["temp_args"][name] = param
```

Comment:
Czemu nie możemy użyć tych parametrów bezpośrednio później, tylko tworzymy PARAMS["temp_args"]?

Reply:
powod taki, ze to jest wczytywane z configa, i dla kazdego configa moze byc inaczej. Zgadzam się, że to może być napisane lepiej. Sugestie appreciated.